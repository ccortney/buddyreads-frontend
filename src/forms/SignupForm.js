import { useState } from "react";
import {useNavigate} from "react-router-dom";
import ErrorAlert from "../common/ErrorAlert";
import { Visibility, VisibilityOff } from '@mui/icons-material';
import {    Typography,
            Grid,  
            Box, 
            TextField, 
            FormControl, 
            InputLabel, 
            OutlinedInput, 
            InputAdornment, 
            Button, 
            IconButton
        } from "@mui/material"

const SignupForm = ({signup}) => {
    const navigate = useNavigate();

    const [formErrors, setFormErrors] = useState([]);
    const [formData, setFormData] = useState({
        email: "", 
        password: "", 
        firstName: "",
        lastName: "",
        profilePicture: "",
        showPassword: false
    });

    const handleClickShowPassword = () => {
        setFormData({
            ...formData, 
            showPassword: !formData.showPassword
        })
    };

    const handleMouseDownPassword = (e) => {
        e.preventDefault();
    };

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(formData => ({...formData, [name]: value}));
    };

    const handleSubmit = async (e) => {
        e.preventDefault();
        delete formData.showPassword;
        let result = await signup(formData);
        if (result.success) {
            navigate('/');
        } else {
            setFormErrors(result.errors);
        }
    }

    return (
        <Grid container align='center' sx={{justifyContent:'center'}}>
            <form onSubmit={handleSubmit}>
 
                <Typography gutterBottom variant="h5" component="div" sx={{pt: 3}}>
                    Create an Account:
                </Typography>
            

            <Box sx={{ display: 'flex', justifyContent:  'center' }}  onSubmit={handleSubmit}>
                <div>
                    <TextField
                        label="Email"
                        id="email"
                        name="email"
                        value={formData.email}
                        onChange = {handleChange}
                        sx={{ m: 1, width: '25ch' }}
                    />
                    <FormControl sx={{ m: 1, width: '25ch' }} variant="outlined">
                        <InputLabel htmlFor="password">Password</InputLabel>
                        <OutlinedInput
                        id="password"
                        type={formData.showPassword ? 'text' : 'password'}
                        name="password"
                        value={formData.password}
                        onChange={handleChange}
                        endAdornment={
                            <InputAdornment position="end">
                            <IconButton
                                aria-label="toggle password visibility"
                                onClick={handleClickShowPassword}
                                onMouseDown={handleMouseDownPassword}
                                edge="end"
                            >
                                {formData.showPassword ? <VisibilityOff /> : <Visibility />}
                            </IconButton>
                            </InputAdornment>
                        }
                        label="Password"
                        />
                    </FormControl>
                </div>  
            </Box>
            <Box >
                <TextField
                    sx={{m: 1, width: 250}}
                    id="firstName"
                    name="firstName"
                    label="First Name"
                    type="text"
                    value={formData.firstName}
                    onChange={handleChange}
                />
                <TextField
                    sx={{m: 1, width: 250}}
                    id="name-input"
                    name="lastName"
                    label="Last Name"
                    type="text"
                    value={formData.lasttName}
                    onChange={handleChange}
                />
            </Box>
            <Box>
                <TextField
                    sx={{m: 1, width: 520}}
                    id="profilePicture"
                    name="profilePicture"
                    label="Profile Picture"
                    type="text"
                    value={formData.profilePicture}
                    onChange={handleChange}
                />
            </Box>
            <Box sx={{display: 'flex', justifyContent: 'right', m: 1}}>
                <Button variant="contained" type="submit">Signup</Button>
            </Box>
            {formErrors.length ? <ErrorAlert severity='error' messages={formErrors}/> : null }
        </form>
    </Grid>
    )
}

export default SignupForm;