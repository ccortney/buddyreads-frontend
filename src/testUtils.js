import UserContext from "./auth/UserContext";

const demoUser = {
  id: 0,
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
