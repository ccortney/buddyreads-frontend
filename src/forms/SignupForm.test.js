import { render } from "@testing-library/react";
import SignupForm from "./SignupForm";
import { NoUserProvider } from "../testUtils";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
          <NoUserProvider>
            <SignupForm />
          </NoUserProvider>
      </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
