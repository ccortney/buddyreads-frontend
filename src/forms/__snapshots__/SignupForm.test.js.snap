// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
  >
    <form>
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-xhhpbl-MuiTypography-root"
      >
        Create an Account:
      </div>
      <div
        class="MuiBox-root css-1l4w6pd"
      >
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root css-ekfslj-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="email"
              id="email-label"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="email"
                name="email"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-r47a1p-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="password"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id="password"
                name="password"
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-ioem3k-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="firstName"
            id="firstName-label"
          >
            First Name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="firstName"
              name="firstName"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  First Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-ioem3k-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="name-input"
            id="name-input-label"
          >
            Last Name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="name-input"
              name="lastName"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Last Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-37z4r7"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Signup
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
